### ISSUE TITLE
Terciary Steam Engines and Accumulators cancel each other out

### ISSUE THREAD

**Issue by megaxiphias** [Dysoch/DyTech#340]
When a terciary steam engine and an accumulator is connected to the same power network, the accumulator won't charge or empty, but all other "generators" (solar panels, primary/secondary steam engines) will go at full power to try to charge the accumulator. When you remove the terciary steam engine from the power network, it all works again.

System: Windows 7 Sp1 64-bit
Factorio 0.12
Dytech Core 1.2.0
Dytech Machine 1.1
Dytech Power 1.1
Dytech War 1.1
Treefarm Lite 0.2.0

**Comment by sistabosse** [comment]
Having the same issue and could not understand what was happening. It's like you say. Having tertiary steam engines and any accumulator will cause the power grid to go haywire and broke my game.

MS Windows 8.1 x64
Factorio 0.12
Dytech Core 1.2.0
Dytech Machine 1.1
Dytech Power 1.1
Dytech War 1.1
Treefarm Lite 0.2.0

**Comment by MagicLegend** [comment]
Correct. This isnt an bug, just strange behaviour of the basegame. Accumulators are tertiary themselves, so have the same priority as the steam engines. If you use primary/secondary steam engines the accumulators -should- charge. Swap your steam engines for an higher priority, and use an different method for your back-up power (use secondary perhaps?) :-)

~ML

**Comment by megaxiphias** [comment]
Ah alright I see, did they change the base game in 0.12 then? Because in 0.11.x there were not a problem with running tertiary steam engines and accumulators in the same grid, it worked perfect!

It sounds kind of odd that the accumulators are tertiary, because they discharge at the same time as the secondary steam engines starts, so you can't having the steam engines running only when the accumulators empty, which you could prior to 0.12 :P

**Comment by MagicLegend** [comment]
I think I'm not understanding your problem correctly. Could you show me an printscreen from your powerinfo (you know, when you click an powerpole) when there are tertiary accumulators and engines in that network? My experience the accumulators just won't charge (with only tertiary's to connected).

Maybe your engines don't produce enough, so the accumulators felt like helping a bit? Nice of them that they do their job ;-)

~ML

**Comment by megaxiphias** [comment]
I don't have that savegame left, but I will try to explain so you can try to replicate the problem, it is quite simple really.

If you have a primary, a secondary and a tertiary steam engine connected in the same powergrid as an accumulator, nothing else, then the primary and the secondary steam engine will try to charge the accumulator at full power, but the accumulator won't carge :p 

To me this feels like a bug, as in 0.11.x this was not the case with accumulators.
Oh and btw, thank you and dysoch for all your work on this mod, it is fantastic!

**Comment by MagicLegend** [comment]
Eh... I need to test that xD It seems like an bug to me as well. I'll look into this tomorrow.

**Comment by MagicLegend** [comment]
So, when playing on my own dytech world, I just realised that I have the same problem there as well. Accumulators just won't charge. For me the engines will not even max out, the accumulator will just show the power shortage (red triangle with the lightning in it) symbol, and do nothing... I'll have to think about this on how I can fix this...

**Comment by HKMontgome42** [comment]
Could it be that the accumulators are trying to provide power while the tertiary is trying to charge the accumulators and the power input/output on the accumulators is different?

**Comment by Berserker66** [comment]
Other mods, such as yuoki, adapted by just not having tertiary steam power.

**Comment by MagicLegend** [comment]
@HKMontgome42 could be. Just use the newest power (1.1.1), then you can change the priority for the engines (high powered engines will come soon)

@Berserker66 I think this is up to the user to figure out. Some people don't use accumulators, and only use tertiary engines for their power supply with primary solars at day. It's just the way that people want to play. I've made the changing really easely with the new GUI, so I think this isn't an problem anymore (you don't get stuck with engines that you don't want to use)

~ML

**Comment by Berserker66** [comment]
If they use primary solar and tertiary steam, then there is no point in tertiary power, is there? They could just as well use secondary steam power and the effect would be identical.

**Comment by MagicLegend** [comment]
@Berserker66 Correct. I'd advise to do this if you want to have solar, steam and accumulator power.
